### PR TITLE
streamingccl: minor refactor

### PR DESCRIPTION
**Pull request by stevendanna**
This pulls out the per-row handling into an interface that we can test
separately.  The goal here is to cut down a bit on conflicts as we
iterate on the batching strategy separately from the underlying insert
queries.

Epic: none
Release note: none

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm: Thanks!

> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125483#-:-O-77qmp2yD9mVC87qvQ:b-g2a5fb)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dt, @navsetlur, and @stevendanna)

___
*[`pkg/ccl/streamingccl/logical/logical_replication_writer_processor.go` line 700 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125483#-O-76jqZ0rRIWBfT9OPi:-O-76jqZ0rRIWBfT9OPj:b6kxn2p) ([raw file](https://github.com/cockroachdb/cockroach/blob/d31e29dff84e0ffa482c6d763d892733b8779824/pkg/ccl/streamingccl/logical/logical_replication_writer_processor.go#L700)):*
> ```Go
> }
> 
> // RowProcessor knows how process a single row from an event stream.
> ```

nit: `s/how process/how to process/`.
___
*[`pkg/ccl/streamingccl/logical/lww_row_processor.go` line 32 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125483#-O-77ITE6dCTQWRluG-Y:-O-77ITE6dCTQWRluG-Z:b-n5f96m) ([raw file](https://github.com/cockroachdb/cockroach/blob/d31e29dff84e0ffa482c6d763d892733b8779824/pkg/ccl/streamingccl/logical/lww_row_processor.go#L32)):*
> ```Go
> )
> 
> // sqlLastWriteWinsRowProcessor is a row handler that implements partial last-write-wins semantics using SQL queries. We assume that the table
> ```

nit: line is too long.
___
*[`pkg/ccl/streamingccl/logical/lww_row_processor.go` line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125483#-O-77Oqb5a-vWNRsEkQG:-O-77Oqb5a-vWNRsEkQH:bu1h90) ([raw file](https://github.com/cockroachdb/cockroach/blob/d31e29dff84e0ffa482c6d763d892733b8779824/pkg/ccl/streamingccl/logical/lww_row_processor.go#L44)):*
> ```Go
> //  1. An UPDATE and a DELETE may be applied out of order because we have no way
> //     from SQL of knowing the write timestamp of the deletion tombstone.
> //  2. The crd_internal_origin_timestamp requires modifying the user's schema.
> ```

nit: `s/crd_internal/crdb_internal/`.

File: pkg/ccl/streamingccl/logical/lww_row_processor.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+// has an crdb_internal_origin_timestamp column defined as:
+//
+//	crdb_internal_origin_timestamp DECIMAL NOT VISIBLE DEFAULT NULL ON UPDATE NULL
+//
+// This row is explicitly set by the INSERT query using the MVCC timestamp of
+// the inbound write.
+//
+// Known issues:
+//
+//  1. An UPDATE and a DELETE may be applied out of order because we have no way
```
Comment (navsetlur) [inline pkg/ccl/streamingccl/logical/lww_row_processor.go:42]: For my own knowledge, how come there isn't an MVCC timestamp for a tombstone?

**Review by navsetlur — APPROVED** [review]
LGTM

**Comment by stevendanna** [comment]
bors r=yuzefovich,navsetlur

**Pull request merged**